src/extension/background-script/actions/allowances/add.ts

```diff
@@ -28,6 +32,14 @@ const add = async (message, sender) => {
       totalBudget: totalBudget,
       remainingBudget: totalBudget,
       lastPaymentAt: 0,
+      createdAt: Date.now().toString(),
+      lnurlAuth: false,
+      payments: [],
```

Comment:
I don't think this can go in the DB. there is no payments array in the DB.
same is for paymentsAmount, paymentsCount, percentage, usedBudget

Reply:
I didn't get this... this would be done for new allowances, so this shouldn't break right?

Reply:
those are the fields of the allowances table:https://github.com/getAlby/lightning-browser-extension/blob/master/src/extension/background-script/db.ts#L15

`allowances.add` is a dexie function that stores data in that table. 

seems we use an "Allowance" type that does not represent the data of the DB but the "allowance"... 
but now when calling `allowances.add` we pass in all the Allowance attributes - and not just the one that are relevant for the DB.

I don't know what dexie does with that, but to me that's wrong.

Reply:
Oh! Got it now! I was thinking we don't save the above things just to save space while working in get action but didn't know we don't even allow them all together in the DB!

Reply:
I guess we can make the types optional `?:` that would make sense as we are using those only when needed.

Reply:
> but I'd say we should use different types a special type for the data stored in the DB

That would make things easier actually...

(Waiting for feedback from them before I proceed)

Reply:
I agree with bumi here. I wasn't aware that the allowances we save in dexie ware not the same that we use withing the app. Unlike blocklist and payments. I split up the type and adjusted the code. Please have a look at this commit:
https://github.com/getAlby/lightning-browser-extension/pull/1044/commits/e4a1fe5d4047eb22c383f9873d618eae82668713